unified-codes/apps/web/src/reducers/ExplorerReducer.ts

```diff
@@ -18,6 +18,22 @@ const initialState = (): IExplorerState => {
       rowsPerPage: 25,
       type: 'medicinal_product',
     },
+    searchBar: {
+      input: '',
+      label: 'Search description',
+    },
+    table: {
+      count: 1,
+      entities: [{ code: 'A', description: 'Drug A', type: 'medicinal_product' }],
+      orderBy: EEntityField.DESCRIPTION,
+      orderDesc: false,
+      rowsPerPage: 25,
+      page: 1,
+    },
+    toggleBar: {
```

Comment:
The more I look at it the more I think this would be nicer as just:

```
toggleBar: {
  enabledTypes: [EEntityType.DRUG],
  disabledTypes: [EEntityType.UNIT_OF_USE, EEntityType.OTHER]
}
```

Thoughts?

Reply:
Good Q, thought about this myself, too!

AFAIK this is redux best practice. As I understand it, we generally want to avoid storing data together which can be separated, as it means its basically atomic and all updates at once (enable one button and they all have to rerender). I also think data structures inside the redux store are generally discouraged as the store should be as flat/denormalised as possible (again, just my understanding, happy to be corrected!).

It's probably most simple to actually have something like: 

```
toggleBar: {
  types: [EEntityType,DRUG, EEntityType.DESCRIPTION, EEntityType.TYPE],
  enabled: [EEntityType.DRUG]
}
```

That way only one field needs to be updated to keep track (so as efficient as `buttons`, as suggested), but without nested state. 

Thoughts on any of above?

Reply:
ok - and I think that either approach is fine really. an array of key value pairs should be fine with individual component updates and rendering. good to explore, and yes - important to consider the shape of the store!